splash/browser_tab.py

```diff
@@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = str(uuid.uuid1())
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class HTMLElement(object):
+    DIMENSIONS_JS_FUNC = """
+        (function(element) {
+            var rect = element.getBoundingClientRect();
+            return {"x":rect.left, "y": rect.top}
+        })(%s)
+    """
+
+    def __init__(self, tab, storage, node):
+        if node == "":
+            raise RuntimeError("Cannot find the requested element")
+
+        self.tab = tab
+        self.storage = storage
+        self.id = node["id"]
+        self.element_js = self.get_element_js()
+
+    def get_element_js(self):
+        return "window['%s']['%s']" % (self.storage.name, self.id)
+
+    def assert_element_exists(self):
+        try:
+            exists = self.tab.evaljs("document.contains(%s)" % self.element_js)
+            if not exists:
+                raise RuntimeError("Element no longer exists in DOM")
+        except JsError:
+            raise RuntimeError("Element no longer exists in DOM")
+
+    def node_property(self, property_name):
+        self.assert_element_exists()
+        return self.tab.evaljs("%s[%s]" % (self.element_js, escape_js(property_name)))
+
+    def node_method(self, method_name):
+        self.assert_element_exists()
+        return self.tab.evaljs("%s[%s]()" % (self.element_js, escape_js(method_name)))
```

Comment:
Should we use `jsfunc` and return the function, so the user can call it with whatever arguments they want? You might need to bind the element to the context or wrap the function for it to work properly.